com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeSubdivisionContext.cs

```diff
@@ -0,0 +1,84 @@
+using UnityEngine.SceneManagement;
+using UnityEngine.Rendering;
+
+namespace UnityEngine.Experimental.Rendering
+{
+    class ProbeSubdivisionContext
+    {
+        public List<(ProbeVolume component, ProbeReferenceVolume.Volume volume)> probeVolumes = new List<(ProbeVolume, ProbeReferenceVolume.Volume)>();
+        public List<(Renderer component, ProbeReferenceVolume.Volume volume)> renderers = new List<(Renderer, ProbeReferenceVolume.Volume)>();
+        public List<(Vector3Int position, ProbeReferenceVolume.Volume volume)> cells = new List<(Vector3Int, ProbeReferenceVolume.Volume)>();
+        public ProbeReferenceVolumeAuthoring refVolume;
+
+        public void Initialize(ProbeReferenceVolumeAuthoring refVolume)
+        {
+            this.refVolume = refVolume;
+            float cellSize = refVolume.cellSizeInMeters;
+
+            foreach (var pv in UnityEngine.Object.FindObjectsOfType<ProbeVolume>())
+            {
+                if (!pv.isActiveAndEnabled)
+                    continue;
+                
+                ProbeReferenceVolume.Volume volume = new ProbeReferenceVolume.Volume(Matrix4x4.TRS(pv.transform.position, pv.transform.rotation, pv.GetExtents()), pv.maxSubdivisionMultiplier, pv.minSubdivisionMultiplier);
+                probeVolumes.Add((pv, volume));
+            }
+
+            // Find all renderers in the scene
+            foreach (var r in UnityEngine.Object.FindObjectsOfType<Renderer>())
+            {
+                if (!r.enabled || !r.gameObject.activeSelf)
+                    continue;
+
+                var flags = GameObjectUtility.GetStaticEditorFlags(r.gameObject);
+                if ((flags & StaticEditorFlags.ContributeGI) == 0)
+                    continue;
+
+                var volume = ProbePlacement.ToVolume(r.bounds);
+
+                renderers.Add((r, volume));
+            }
+
+            // Generate all the unique cell positions from probe volumes:
+            var refVolTransform = ProbeReferenceVolume.instance.GetTransform();
+            var cellTrans = Matrix4x4.TRS(refVolTransform.posWS, refVolTransform.rot, Vector3.one);
+            HashSet<Vector3Int> cellPositions = new HashSet<Vector3Int>();
+            foreach (var pv in probeVolumes)
+            {
+                var probeVolume = pv.component;
+                var halfSize = probeVolume.size / 2.0f;
+                var minCellPosition = (probeVolume.transform.position - halfSize) / cellSize;
+                var maxCellPosition = (probeVolume.transform.position + halfSize) / cellSize;
+
+                Vector3Int min = new Vector3Int(Mathf.FloorToInt(minCellPosition.x), Mathf.FloorToInt(minCellPosition.y), Mathf.FloorToInt(minCellPosition.z));
+                Vector3Int max = new Vector3Int(Mathf.CeilToInt(maxCellPosition.x), Mathf.CeilToInt(maxCellPosition.y), Mathf.CeilToInt(maxCellPosition.z));
+
+                for (int x = min.x; x < max.x; x++)
```

Comment:
I might very well be missing something, but given we do this cellPositions additions per probes, if we have two probe volumes touching the same cell, won't we be adding two cell positions for the same cell here?

Reply:
It's fine because the add is protected by the HashSet `cellPositions`, so it's not possible to have duplication in the final list